src/engine/SCons/Script/SConsOptions.py

```diff
@@ -28,7 +28,7 @@
 import sys
 import textwrap
 
-no_hyphen_re = re.compile(r'(\s+|(?<=[\w\!\"\'\&\.\,\?])-{2,}(?=\w))')
+no_hyphen_re = re.compile(r'(\s+|(?<=[\w!\"\'&.,?])-{2,}(?=\w))')
```

Comment:
A note to review this as well. I'm not sure pycharm's regex improvers are correct..